scripts/generate_matrix.py

```diff
@@ -0,0 +1,273 @@
+# ruff: noqa: TRY003
+import importlib.metadata
+import json
+import sys
+import urllib.error
+import urllib.request
+from functools import lru_cache
+import typing
+
+from packaging.specifiers import SpecifierSet
+from packaging.version import Version, parse
+
+# Constants
+CACHE_SIZE = 128  # Adjust as needed
+PACKAGE_NAME = "scipy"
+DEPENDENCY_NAME = "numpy"
+MINIMUM_DEPENDENCY_VERSION_FOR_PYTHON_3_12 = Version("1.26.0")
+
+
+class FileInfo(typing.TypedDict, total=False):
+    filename: str
+    arch: str
+    platform: str
+    platform_version: str | None
+    download_url: str
+    requires_python: str
+
+
+class Release(typing.TypedDict):
+    version: str
+    stable: bool
+    release_url: str
+    files: list[FileInfo]
+
+
+class PackageVersions(typing.TypedDict, total=False):
+    releases: dict[str, list[FileInfo]]
+
+
+@lru_cache(maxsize=CACHE_SIZE)  # type: ignore[no-any-expr]
```

Comment:
wouldn't a `functools.cache` be enough in this case?

Reply:
![image](https://github.com/user-attachments/assets/363e8d04-46ba-412e-b152-2d16050fccb6)

once again... I blame GPT, I didn't write that code I just told it to use caching ;)